beacon-chain/blockchain/service.go

```diff
@@ -300,6 +300,8 @@ func (c *ChainService) processBlock(block *types.Block) error {
 		return errors.New("Block failed validity conditions")
 	}
 
+	aState = aState.UpdateAttestations(block.Attestations())
```

Comment:
There's a weird issue during cycle transition that I'll try and explain:

During a cycle transition, `crystallized_state` and `active_state` are updated sequentially. However, since `crystallized_state` is updated first, its updated with a stale `active_state`. The result is that attestations for that block are never processed. I fixed this by appending attestations _before_ updating `crystallized_state`.